### ISSUE TITLE
Add setAllowWrites as a config file option

### ISSUE THREAD

**Issue by chrisbrowning** [graphhopper/graphhopper#2815]
We are users of the docker image for and would like to have a config file option to run our graphhopper server in "read-only mode". Our forked implementation of this is simple, in the `GraphHopper.java` file we have:

```java
allowWrites = ghConfig.getBool("graph.allow_writes", allowWrites);
```

Keeping our fork updated can be difficult for such a small feature and we'd prefer to stop using our fork and start using the graphhopper repo directly, so we would be grateful if something like this could be implemented in the graphhopper repo.

**Comment by easbar** [comment]
Wouldn't that be the same as using `graph.dataaccess.default_type: MMAP_RO`?

**Comment by chrisbrowning** [comment]
I don't think so as in order to avoid attempting a read lock (we are using a read only volume) it seems that allow writes needs to be [explicitly disabled](https://github.com/graphhopper/graphhopper/blob/master/core/src/main/java/com/graphhopper/GraphHopper.java#L1022-L1026).

**Comment by easbar** [comment]
You mean `GraphHopper#allowWrites` can be true even when `DAType#allowWrites` is false?

**Comment by chrisbrowning** [comment]
We tried setting the configuration value that you suggested might work and got the following error:
`To avoid reading partial data we need to obtain the read lock but it failed`

This appears to occur when graphhopper tries setting a lock on a read-only filesystem. When `allowWrites` is set to false, it later sets `graph.dataaccess.default_type` =  `MMAP_RO`, but by manually setting the `default_type` we are not disabling `allowWrites` so graphhopper still tries to set a lock [here](https://github.com/graphhopper/graphhopper/blob/master/core/src/main/java/com/graphhopper/GraphHopper.java#L1009).

**Comment by Nakaner** [comment]
> Wouldn't that be the same as using `graph.dataaccess.default_type: MMAP_RO`?

`MMAP_RO` implies that it works in a read-only directory. However, the check if the directory is writeable (lockable) [happens](https://github.com/graphhopper/graphhopper/blob/5eb09e78a7216b7a436f6f592e1c6ee1bc4c9d6f/core/src/main/java/com/graphhopper/GraphHopper.java#L1032) before data access is initialized (in `BaseGraph.Builder` class).

**Comment by knowname** [comment]
> We tried setting the configuration value that you suggested might work and got the following error: `To avoid reading partial data we need to obtain the read lock but it failed`
> 
> This appears to occur when graphhopper tries setting a lock on a read-only filesystem. When `allowWrites` is set to false, it later sets `graph.dataaccess.default_type` = `MMAP_RO`, but by manually setting the `default_type` we are not disabling `allowWrites` so graphhopper still tries to set a lock [here](https://github.com/graphhopper/graphhopper/blob/master/core/src/main/java/com/graphhopper/GraphHopper.java#L1009).


As a workaround you could place a softlink to a writeable location next to the graph: `ln -s /tmp/gh.lock gh.lock`
